aten/src/ATen/native/cuda/DistanceKernel.cu

```diff
@@ -157,6 +157,33 @@ __global__ static void pdist_backward_kernel_cuda_impl(scalar_t * buffer, const
   }
 }
 
+template <typename scalar_t, typename F>
+__global__ static void cdist_kernel_cuda_impl(scalar_t * result, const scalar_t * x1, const scalar_t * x2, const scalar_t p, const int64_t r1, const int64_t r2, const int64_t m) {
+
```

Comment:
since this isn't implemented yet (which is fine!) can you just kill the cuda changes?

Note you can make sure this only executes on CPU by adding a `CPU:` only dispatch to this function in native_functions.yaml such as https://github.com/pytorch/pytorch/blob/07a090247a4d82d1cde724409ef152d7e8ad99f6/aten/src/ATen/native/native_functions.yaml#L518.

Reply:
Checked these testcases:

1) CPU and CUDA devices
>>> import torch;
>>> torch.cuda.is_available()
True
>>> x = torch.randn((3, 5), device='cpu')
>>> y = torch.randn((4, 5), device='cuda')
>>> torch.cdist(x, y, 2)
Segmentation fault (core dumped)

2) Different CUDA devices
>>> import torch;
>>> print(torch.cuda.current_device())
0
>>> x = torch.randn((3, 5), device='cuda')
>>> x
tensor([[ 1.4912,  0.0627,  0.2258, -0.0317, -1.5003],
        [-0.9519, -1.0205,  1.0646,  0.0418, -0.6407],
        [ 0.4870,  1.4298, -0.3464, -1.5242, -1.2492]], device='cuda:0')
>>> torch.cuda.set_device(1)
>>> y = torch.randn((4, 5), device='cuda')
>>> y
tensor([[ 1.2308, -1.4090,  0.8967, -1.4265, -0.0454],
        [-0.5917, -0.6217,  1.0569, -1.4225, -1.2125],
        [ 1.0328, -0.6643, -0.2227, -1.4387,  1.5642],
        [-2.7521,  0.3133,  0.3392, -0.9948,  0.0224]], device='cuda:1')
>>> torch.cdist(x, y, 1)
THCudaCheck FAIL file=/data/users/ifedan/pytorch/aten/src/THC/THCReduceAll.cuh line=317 error=77 : an illegal memory access was encountered

So I added correspondent check to validate that CUDA devices have same index